File: src/main/java/com/patres/timetable/service/dto/LessonDTO.java
package com.patres.timetable.service.dto;


import javax.validation.constraints.*;
import java.io.Serializable;
import java.util.Objects;

public class LessonDTO extends AbstractDivisionOwnerDTO implements Serializable {

    @NotNull
    private String name;

    @NotNull
    private Long startTime;

    @NotNull
    private Long endTime;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Long getStartTime() {
        return startTime;
    }

    public void setStartTime(Long startTime) {
        this.startTime = startTime;
    }

    public Long getEndTime() {
        return endTime;
    }

    public void setEndTime(Long endTime) {
        this.endTime = endTime;
    }


    @Override
    public String toString() {
        return "LessonDTO{" +
            "id=" + getId() +
            ", name='" + getName() + "'" +
            ", startTime='" + getStartTime() + "'" +
            ", endTime='" + getEndTime() + "'" +
            "}";
    }
}

File: src/main/kotlin/com/patres/timetable/service/dto/LessonDTO.kt
package com.patres.timetable.service.dto

import java.io.Serializable
import javax.validation.constraints.NotNull

class LessonDTO(

    @get:NotNull
    var name: String? = null,

    @get:NotNull
    var startTimeString: String? = null,

    @get:NotNull
    var endTimeString: String? = null,

    divisionOwnerId: Long? = null

) : AbstractDivisionOwnerDTO(divisionOwnerId = divisionOwnerId), Serializable
